src/cli/argument_default_value_from_config_provider.py

```diff
@@ -0,0 +1,21 @@
+from typing import Any, Optional
+
+from src.cli.command import Command
+from src.utils import Project
+
+
+class ArgumentDefaultValueFromConfigProvider:
+    def __init__(self, project: Project) -> None:
+        self._project = project
+        super().__init__()
+
+    def get_default_value(
+        self, command: Optional[Command], argument: Command.Argument
```

Comment:
I think it should be 
```suggestion
        self, command: Command, argument: Command.Argument
```
We should never request data from shared config it should be only a fallback. It will make interface more defensive

Reply:
Also we should have separate section for root parameters. Keeping them inside shared config is misleading

Reply:
We can make two methods
```
def get_command_config_value( self, command: Command, argument: Command.Argument)
def get_root_config_value( self, argument: Command.Argument)
```
Interface would be simpler and more defensive

Reply:
Is it misleading? Root args (--no-color) are shared across commands. The only exception is the `--version`, which behaves like command.

> We should never request data from shared config it should be only a fallback. It will make interface more defensive
I don't get it. It's already used as a fallback.

Reply:
Giving it afterthought leave it as it is